modules/trusted-firmware-m/Kconfig

```diff
@@ -58,15 +58,59 @@ config TFM_EXPERIMENTAL
 	select EXPERIMENTAL
 	default y if !TFM_PROFILE_TYPE_MINIMAL
 
+if TFM_PARTITION_CRYPTO
+
 config TFM_CRYPTO_ENGINE_BUF_SIZE
 	int
-	prompt "TFM Crypto engine buffer size" if !TFM_PROFILE_TYPE_MINIMAL
+	prompt "TF-M Crypto - Engine buffer size" if !TFM_PROFILE_TYPE_MINIMAL
 	default 1 if TFM_PROFILE_TYPE_MINIMAL
 	default 12288 if PSA_CRYPTO_DRIVER_ALG_RSA_OAEP_CC3XX || \
 		PSA_CRYPTO_DRIVER_ALG_RSA_PKCS1V15_CRYPT_CC3XX || \
 		PSA_CRYPTO_DRIVER_ALG_RSA_PKCS1V15_SIGN_CC3XX || \
 		PSA_CRYPTO_DRIVER_ALG_RSA_PSS_CC3XX
 	default 8320
+	help
+	  Buffer used by Mbed Crypto for its own allocations at runtime.
+	  This is a buffer allocated in static memory.
+	  To be configured based on the desired use case and application requirements.
+
+config TFM_CRYPTO_CONC_OPER_NUM
+	int
+	prompt "TF-M Crypto - Number of concurrent operations" if !TFM_PROFILE_TYPE_MINIMAL
+	default 1 if TFM_PROFILE_TYPE_MINIMAL
+	default 8
+	help
+	  This parameter defines the maximum number of possible
+	  concurrent operation contexts (cipher, MAC, hash and key derivation)
+	  for multi-part operations, that can be allocated simultaneously
+	  at any time. To be configured based on the desired use case and
+	  application requirements.
+
+config TFM_CRYPTO_IOVEC_BUFFER_SIZE
+	int
+	prompt "TF-M Crypto - PSA FF IO vector buffer size" if !TFM_PROFILE_TYPE_MINIMAL
+	depends on TFM_IPC
+	default 16384 if TFM_REGRESSION_S || TFM_REGRESSION_NS
```

Comment:
systems shouldn't know of their users.

Can we get TFM_REGRESSION_S and TFM_REGRESSION_NS to configure this option explicitly ?

I can't recall exactly what TFM_REGRESSION_S and TFM_REGRESSION_NS are. If they are samples then they have prj.conf files don't they?

Reply:
@tejlmand : Have you seen a case where a Zephyr sample uses one of our systems and we need the sample to configure our system off of the default values?

Any ideas?

If it were a downstream sample we would just use prj.conf.

Reply:
Can we calculate the buffer size based on other enabled options?

Reply:
Why does the buffer size need to be 16k for the regression sample, what specifically is using that RAM ?

Reply:
I suggest we upstream a patch to one of these files:

ls /home/sebo/ncs/modules/tee/tf-m/tf-m-tests/test/config/
check_config.cmake            default_s_test_config.cmake  profile
default_ns_test_config.cmake  default_test_config.cmake    set_config.cmake

to configure the TFM_CRYPTO_IOVEC_BUFFER_SIZE appropriately when the regression test is run.

It is the regression test that knows that it uses a lot of RAM and needs to configure the stuff it uses appropriately.

Libraries should not know of samples that have high ram usage and configure themselves accordingly.

Reply:
This value was not required by this PR, and so I will remove the setting of this value here and only add the configuration.
We can deal with the value required by the test in the PR that has the matching change.